binarySearch/binarySearch.js

```diff
@@ -10,6 +10,49 @@
  *	* console.log(index); // 1
 **/
 
-const binarySearch = (nums, target) => {
+// // version 1: indexOf (gets the job done, but misses the point entirely)
+// const binarySearch = (nums, target) => {
+//   return nums.indexOf(target);
+// };
+
+// version 2: Divide and Conquer √
+
+// I: A SORTED array (low to high) of integers AND a target number
+// O: the array index of the target
+  //  we are presuming the array actually contains target
 
+/* FUNCTION:
+  * We know the target is at one of the array indexes. Take the average of min & max range, divide by half
+    * If target number at that index, return index #
+    * Is the value at that index   less  than the target? Then increase the minimum to the guess index position + 1
+    * Is the value at that index greater than the target? Then decrease the maximum to the guess index position - 1
+*/
+
+const binarySearch = (nums, target) => {
+  if (nums.length < 1) return "Error001: empty array";
+  if (nums.indexOf(target) === -1) return "Error002: target number not in array";
+  let min = 0;
+  let max = nums.length - 1;
+  let guess;
+  while (min <= max) {
+    guess = Math.floor((min + max) / 2);
+    if (nums[guess] === target) { return guess }
+    else if (nums[guess] < target) { min = guess + 1 }
+    else { max = guess - 1 }
+  }
 };
+
+// // TEST SUITE
```

Comment:
I love that you do this! It's a solid way to demonstrate your code and how it's operating/running Great work!